e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java

```diff
@@ -0,0 +1,434 @@
+import org.junit.jupiter.api.*;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.*;
+
+@Testcontainers
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class TestRunner {
+
+    @Container
+    public static DockerComposeContainer container = new DockerComposeContainer(
+            new File("../docker-e2e-compose.yml"))
+            .withExposedService("db", 5432)
+            .withExposedService("api", 8080).withPull(false).withLocalCompose(true);
+
+    private static HttpClient httpClient;
+
+    private static DriverManagerDataSource dataSource;
+
+    private static String AB2D_API_URL = "http://localhost:8080/api/v1/fhir/";
+
+    private static String AB2D_ADMIN_URL = "http://localhost:8080/api/v1/admin/";
+
+    private static final String PATIENT_EXPORT_PATH = "Patient/$export";
+
+    private static String jwtStr = null;
+
+    private static String TEST_USER;
+
+    private static final long DEFAULT_TIMEOUT = 30;
+
+
+    @BeforeAll
+    public void init() throws IOException, InterruptedException, JSONException {
+        container.start();
+
+        Yaml yaml = new Yaml();
+        InputStream inputStream = getClass().getResourceAsStream("/default-config.yml");
+        Map<String, String> yamlMap = yaml.load(inputStream);
+        String oktaUrl = yamlMap.get("okta-url");
+        TEST_USER = yamlMap.get("username");
+        String password = yamlMap.get("password");
+        AB2D_API_URL = yamlMap.get("ab2d-api-url");
+        AB2D_ADMIN_URL = yamlMap.get("ab2d-admin-url");
+
+        httpClient = HttpClient.newBuilder()
+                .version(HttpClient.Version.HTTP_1_1)
+                .followRedirects(HttpClient.Redirect.NORMAL)
+                .build();
+
+        var jwtRequestParms = new HashMap<>() {{
+            put("grant_type", "password");
+            put("username", TEST_USER);
+            put("password", password);
+            put("scope", "openid");
+        }};
+
+        HttpRequest jwtRequest = HttpRequest.newBuilder()
+                .uri(URI.create(oktaUrl))
+                .timeout(Duration.ofSeconds(DEFAULT_TIMEOUT))
+                .header("Content-Type", "application/x-www-form-urlencoded")
+                .header("Accept", "application/json")
+                .header("Authorization", "Basic MG9hMXB4YXB6ZE9XaUhmOXUzNTc6WGIxQnpDb1ktS0Y4Z1FhTF9JMmVzVmxsdkpZZ2VzX2d1QTVHZTlpaQ==")
```

Comment:
It may be more sustainable to pass in the clientId and clientPassword in the yaml and then construct the Basic Auth value "Basic Base64(clientId:clientPassword). I have those values if you need them (I have decoded them)

Reply:
👍